### PR TITLE
(PUP-3921) Add 404 error suggesting an upgrade

### PR DESCRIPTION
**Pull request by fpringvaldsen**
Add a 404 error that is triggered when requests are made to
routes without the /puppet or /puppet-ca prefixes to indicate
to the user that they may need to upgrade Puppet on their
agents.

This is a new PR addressing the same ticket as https://github.com/puppetlabs/puppet/pull/3532, which I am closing in favor of this one.

### CODE REVIEW

File: lib/puppet/network/http.rb [hunk 1]
```diff
@@ -5,6 +5,9 @@ module Puppet::Network::HTTP
   MASTER_URL_PREFIX = "/puppet"
   CA_URL_PREFIX = "/puppet-ca"
 
+  PUPPET_API_VERSIONS = "v3, v2"
```
Comment (kylog) [inline lib/puppet/network/http.rb]: I could be behind the times but I thought puppet 4 was `v3`only for the `puppet` part of the API?

Comment (rlinehan) [inline lib/puppet/network/http.rb]: Yep, this is correct. See PUP-3581.

**Comment by fpringvaldsen** [comment]
@jeffmccune @rlinehan I've addressed your feedback on https://github.com/puppetlabs/puppet/pull/3532 with this PR.

File: lib/puppet/network/http.rb (hunk 1 above)
Comment (fpringvaldsen) [inline lib/puppet/network/http.rb]: Ah, I see. I knew v2 was getting removed but wasn't sure if that was going to be removed for Puppet 4 or not. I've fixed it to only say v3.

File: lib/puppet/network/http.rb [hunk 2]
```diff
@@ -5,6 +5,9 @@ module Puppet::Network::HTTP
   MASTER_URL_PREFIX = "/puppet"
   CA_URL_PREFIX = "/puppet-ca"
 
+  PUPPET_API_VERSIONS = "v3"
+  PUPPET_CA_API_VERSIONS = "v1"
+
```
Comment (kylog) [inline lib/puppet/network/http.rb]: Would it make sense to name these two VERSIONS constants to match up with the corresponding `_URL_PREFIX` constants above? I.e. to make it more obvious that they go with those.

E.g.

```
MASTER_URL_PREFIX = "/puppet"
MASTER_URL_VERSIONS = "v3"

CA_URL_PREFIX = "/puppet-ca"
CA_URL_VERSIONS = "v1"
```

Naming is hard, so please feel free to ignore me :)

Comment (nfagerlund) [inline lib/puppet/network/http.rb]: +1 to that.

**Comment by fpringvaldsen** [comment]
@kylog I've renamed the constants as per your suggestion

**Comment by kylog** [comment]
Oops, I meant to circle back to this one. I'm :+1: 

@jeffmccune or @rlinehan or @nfagerlund you all had comments on the first PR. Thoughts?

**Comment by rlinehan** [comment]
:+1:

**Pull request merged**